tests/test_kuzmin.py

```diff
@@ -136,23 +137,97 @@ def test_K1_function_of_temperature():
     assert k1_entity == k1
 
 
-def test_kuzmin_properties():
-    """Test the kuzmin_properties function."""
-    Ms = me.Ms([200, 100.0], unit=u.A / u.m)
-    Tc = me.Entity("ThermodynamicTemperature", value=[0, 100], unit="K")
+def test_kuzmin_properties_all_info():
+    """Test the kuzmin_properties function with all information."""
+    Tc = me.Tc(value=500, unit="K")
     K1_0 = me.Ku(1e5, unit=u.J / u.m**3)
-    result = kuzmin_properties(Ms, Tc, K1_0)
+    T_data = me.Entity("ThermodynamicTemperature", value=[0, 100, 200, 300, 400, 500])
+    Ms_0 = me.Ms(100)
+    Ms_data = me.Ms(kuzmin_formula(Ms_0=Ms_0, T_c=Tc, s=0.75, T=T_data))
+    result = kuzmin_properties(Ms=Ms_data, T=T_data, Tc=Tc, Ms_0=Ms_0, K1_0=K1_0)
     assert isinstance(result, KuzminResult)
     assert isinstance(result.Ms, _Ms_function_of_temperature)
     assert isinstance(result.A, _A_function_of_temperature)
     assert isinstance(result.K1, _K1_function_of_temperature)
     assert isinstance(result.Tc, me.Entity)
     assert isinstance(result.s, u.Quantity)
-
-    result = kuzmin_properties(Ms, Tc)
+    assert result.Tc == Tc
+    assert result.K1(0) == K1_0
+    assert np.allclose(result.s, 0.75)
+    assert result.Ms(T_data) == Ms_data
+    assert result.Ms(0) == Ms_0
+    Tc = me.Tc(value=[500], unit="K")
+    K1_0 = me.Ku([1e5], unit=u.J / u.m**3)
+    Ms_0 = me.Ms([100])
+    Ms_data = me.Ms(kuzmin_formula(Ms_0=Ms_0, T_c=Tc, s=0.75, T=T_data))
+    result = kuzmin_properties(Ms=Ms_data, T=T_data, Tc=Tc, Ms_0=Ms_0, K1_0=K1_0)
+    assert result.Tc == me.Tc(500)
```

Comment:
I assume we could use
```suggestion
    assert result.Tc == Tc
```

Reply:
https://github.com/MaMMoS-project/mammos-analysis/blob/1d8a51332943420e0513126dbd9ad5c6c31b7caf/tests/test_kuzmin.py#L159

No, we cannot. `Tc` is a 1-d vector and `result.Tc` will be a float quantity.